### ISSUE TITLE
When I select "Cancel", many sites don't load.

### ISSUE THREAD

**Issue by chesternick** [abrahamjuliot/ublock-origin_abort_browser_fingerprinting#6]
When you select "Cancel", many sites do not load.

Can you add a function to substitute random values for all blocked parameters?

Your script is very good, and if you add this feature, it will be very good.

**Comment by abrahamjuliot** [comment]
Thank you. I will give this some thought. 

`cancel` is intended to attack the script (abort if possible) by throwing a storm of random errors. This is mostly intended to shut down untrusted or suspicious activity. Site breakage is intended. To feed the script only random values, select `ok`.

**Comment by chesternick** [comment]
When I click "ok" it is not replaced by random. 
Once, a random GPU model was changed on some browser. But the other parameters are not substituted for random ones.
On your SUPER.site (https://abrahamjuliot.github.io/creepjs/) , the "Math" value is highlighted in red, but it matches the actual value. This is in the mode when "ok" is selected.

**Comment by abrahamjuliot** [comment]
Give it a test at these sites:
- https://fingerprintjs.com/demo
- https://amiunique.org/fp
- https://panopticlick.eff.org

`creepjs` is currently too difficult for me to overcome, but there's a lot I can clean up here to not get a negative F score. On that note, I'm not aware of any extensions, addons, or browsers that effectively randomize the fingerprint on `creepjs`. That's surely possible using built in methods: chrome's sensor API (time zone), built in device emulator, puppeteer automation, or a script attack.

The `creepjs` test site uses offscreen canvas (if supported) to fingerprint webgl. I plan to cover that here eventually, but it's still [experimental](https://developer.mozilla.org/en-US/docs/Web/API/OffscreenCanvas) with limited browser support. I would not expect it to be used in the wild, but I can add this protection. 

If `creepjs` detects prototype lies or mathematical lies, the section hash turns red and the metrics are removed from the stable fingerprint. In this case, prototype lies are detected (the lie is the watch function added to the prototype), but I'm not randomizing the math computation.

**Comment by mendelsphotography** [comment]
on this note can you make that you can somehow go back Let's say you decide to deny the site and it breaks it so you want to go back and allow it currently you can not, but is it possible to make it that you can? @abrahamjuliot

**Comment by abrahamjuliot** [comment]
Yes, going back within the current tab is currently not an option, but you may start fresh by opening the site URL in a new tab.

**Comment by mendelsphotography** [comment]
> Yes, going back within the current tab is currently not an option, but you may start fresh by opening the site URL in a new tab.

tried that and found it did not always fix

**Comment by abrahamjuliot** [comment]
That is odd. Just make sure to not open a new tab in the background. After a short delay, it will default to cancel. That's not by design, but rather the downside of the alert message.

On a side note, if you are interested in something a little more light weight and less aggressive, use this in your custom filters. It uses UBO's built in `aopr` method to abort on property reads and sends a random error message. 

However, note that unique/random errors can be traced. I only recommend it as the default for sites you don't trust.

```
*##+js(aopr, HTMLCanvasElement.prototype.toDataURL)
*##+js(aopr, WebGLRenderingContext.prototype.getParameter)
```

**Comment by mendelsphotography** [comment]
Oh ok i think that explains why then The fact that sometimes was opening right after so did not clear yet but then sometimes it went back

**Comment by mendelsphotography** [comment]
thanks for the explantion
